browser/ui/webui/ai_chat/ai_chat_untrusted_conversation_ui.cc

```diff
@@ -353,6 +355,23 @@ AIChatUntrustedConversationUI::AIChatUntrustedConversationUI(
   source->OverrideContentSecurityPolicy(
       network::mojom::CSPDirectiveName::ScriptSrc,
       "script-src 'self' chrome-untrusted://resources;");
+
+  // If the feature is not enabled then don't add the origin to the CSP.
+  if (base::FeatureList::IsEnabled(ai_chat::features::kRichSearchWidgets)) {
+    source->OverrideContentSecurityPolicy(
+        network::mojom::CSPDirectiveName::FrameSrc,
+        base::StrCat({"frame-src 'self' ",
+                      ai_chat::features::kRichSearchWidgetsOrigin.Get(), ";"}));
```

Comment:
@ShivanKaul @thypon @petemill I'm a bit unsure about doing this for the origin. WDYT?

Reply:
actually it'd be better to match the exact path `/embed.html` instead of the entire origin: https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Content-Security-Policy#host-source